### PR TITLE
Specify base OCI runtime spec

### PR DESCRIPTION
**Pull request by mxpv**
This PR adds `base_runtime_spec` to runtime config to use a file with base OCI spec.

Details: https://github.com/containerd/cri/issues/1488

Signed-off-by: Maksym Pavlenko <pavlenko.maksym@gmail.com>

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -135,6 +135,11 @@ version = 2
       # i.e pass host devices through to privileged containers.
       privileged_without_host_devices = false
 
+      # base_runtime_spec is a file path to a JSON file with the OCI spec that will be used as the base spec
+      # that all container's are created from.
+      # Use containerd's `ctr oci default-spec > /etc/containerd/cri-base.json` to output initial spec file.
+      base_runtime_spec = ""
```
Comment (mikebrow) [inline docs/config.md:141]: nit.. may want to mention a restart of containerd is required to load changes to the one default spec...
still running containers and restarted containers will still be using the original spec from which that container was created.

Comment (mxpv) [reply docs/config.md:143]: Done

File: pkg/server/container_create.go [hunk 2]
```diff
@@ -297,12 +297,33 @@ func (c *criService) volumeMounts(containerRootDir string, criMounts []*runtime.
...
+		if !ok {
+			return nil, errors.Errorf("can't find base OCI spec %q", baseSpecFile)
+		}
+
+		spec := oci.Spec{}
+		if err := util.DeepCopy(&spec, &baseSpec); err != nil {
+			return nil, errors.Wrap(err, "failed to clone OCI spec")
+		}
+
+		if err := oci.ApplyOpts(ctx, nil, container, &spec, opts...); err != nil {
```
Comment (crosbymichael) [inline pkg/server/container_create.go:316]: After doing this you need to make sure that you setup the cgrouppath based on the namespace of context.

```go
				var s oci.Spec
		if err := json.NewDecoder(f).Decode(&s); err != nil {
			return errors.Wrap(err, "decode spec")
		}
		s.Linux.CgroupsPath = filepath.Join("/", ns, c.ID)

		if err := oci.ApplyOpts(ctx, client, c, &s, opts...); err != nil {
			return errors.Wrap(err, "apply opts")
		}
```

Comment (mxpv) [reply pkg/server/container_create.go]: Good catch, thank you.

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**